src/core/server/plugins/plugins_system.ts

```diff
@@ -158,13 +158,41 @@ export class PluginsSystem {
         timeout: 30 * Sec,
         errorMessage: `Start lifecycle of "${pluginName}" plugin wasn't completed in 30sec. Consider disabling the plugin and re-start.`,
       });
-
       contracts.set(pluginName, contract);
     }
+    await this.healthCheckOpenSearchPlugins(deps);
 
     return contracts;
   }
 
+  private async healthCheckOpenSearchPlugins(deps: PluginsServiceStartDeps) {
+    // make _cat/plugins?format=json call to the OpenSearch instance
+    const opensearchPlugins = await this.getOpensearchPlugins(deps);
```

Comment:
Do we need to handle the failure case for this call? Or does `getOpensearchPlugins` need to return an empty array or other fallback value if it fails?

Reply:
Yes that's a good call. We can return an empty array in this case which will eventually only throw error if there is a plugin dependency defined on backend plugins.